alarms.tf

```diff
@@ -69,35 +69,35 @@ locals {
 }
 
 resource "aws_cloudwatch_log_metric_filter" "default" {
-  count          = "${length(local.filter_pattern)}"
+  count          = length(local.filter_pattern)
   name           = "${local.metric_name[count.index]}-filter"
-  pattern        = "${local.filter_pattern[count.index]}"
-  log_group_name = "${var.log_group_name}"
+  pattern        = local.filter_pattern[count.index]
+  log_group_name = var.log_group_name
 
   metric_transformation {
-    name      = "${local.metric_name[count.index]}"
-    namespace = "${local.metric_namespace}"
-    value     = "${local.metric_value}"
+    name      = local.metric_name[count.index]
+    namespace = local.metric_namespace
+    value     = local.metric_value
   }
 }
 
 resource "aws_cloudwatch_metric_alarm" "default" {
-  count               = "${length(local.filter_pattern)}"
+  count               = length(local.filter_pattern)
   alarm_name          = "${local.metric_name[count.index]}-alarm"
   comparison_operator = "GreaterThanOrEqualToThreshold"
   evaluation_periods  = "1"
-  metric_name         = "${local.metric_name[count.index]}"
-  namespace           = "${local.metric_namespace}"
-  period              = "300"                                                                         // 5 min
+  metric_name         = local.metric_name[count.index]
+  namespace           = local.metric_namespace
+  period              = "300" // 5 min
   statistic           = "Sum"
   treat_missing_data  = "notBreaching"
-  threshold           = "${local.metric_name[count.index] == "ConsoleSignInFailureCount" ? "3" :"1"}"
-  alarm_description   = "${local.alarm_description[count.index]}"
-  alarm_actions       = ["${local.endpoints}"]
+  threshold           = "${local.metric_name[count.index] == "ConsoleSignInFailureCount" ? "3" : "1"}"
```

Comment:
Minor one here:
-  `"${local.metric_name[count.index] == "ConsoleSignInFailureCount" ? "3" : "1"}"`
For tf > 0.12 should ideally be:
- `local.metric_name[count.index] == "ConsoleSignInFailureCount" ? "3" : "1"`

Reply:
### Where? 
 `/examples/simple/main.tf`

#### 1. Warning: Quoted references are deprecated

> In this context, references are expected literally rather than in quotes.
> Terraform 0.11 and earlier required quotes, but quoted references are now
> deprecated and will be removed in a future version of Terraform. Remove the
> quotes surrounding this reference to silence this warning.

```terraform
  on main.tf line 107, in resource "aws_cloudtrail" "default":
 107:   depends_on                    = ["aws_s3_bucket_policy.default", "aws_iam_role_policy.policy"]
```
Updated code:
```terraform
resource "aws_cloudtrail" "default" {
  name                          = "cloudtrail-${var.region}"
  enable_logging                = "true"
  s3_bucket_name                = aws_s3_bucket.default.id
  enable_log_file_validation    = "false"
  is_multi_region_trail         = "true"
  include_global_service_events = "true"
  cloud_watch_logs_role_arn     = aws_iam_role.cloudtrail_cloudwatch_events_role.arn
  cloud_watch_logs_group_arn    = aws_cloudwatch_log_group.default.arn
  depends_on                    = [aws_s3_bucket_policy.default, aws_iam_role_policy.policy]
}
```

---

#### 2. Error: Unsupported argument

```
  on main.tf line 87, in data "aws_iam_policy_document" "assume_policy":
  87:     principals = {
```

An argument named "principals" is not expected here. Did you mean to define a
block of type "principals"?

Updated Code:

```terraform
data "aws_iam_policy_document" "assume_policy" {
  statement {
    effect  = "Allow"
    actions = ["sts:AssumeRole"]

    principals {
      type        = "Service"
      identifiers = ["cloudtrail.amazonaws.com"]
    }
  }
}
```